amplify/data/resource.ts

```diff
@@ -23,7 +23,7 @@ const schema = a.schema({
       team: a.belongsTo("Team", "teamId"),
```

Comment:
On line 22, we need to add a authorization rule to the field to allow deleting. Otherwise when we set it to null when someone leaves a team, there's an Unauthroized failure since we are technically deleting that field.
```suggestion
      teamId: a
        .id()
        .authorization((allow) =>
          allow.owner().to(["update", "read", "delete"]),
        ),
```